app/services/integration_customers/create_service.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module IntegrationCustomers
+  class CreateService < BaseService
+    def initialize(params:, integration:, customer:)
+      @customer = customer
+      super(params:, integration:)
+    end
+
+    def call
+      result = super
+      return result if result.error
+
+      res = if sync_with_provider
```

Comment:
What should be the flow if `sync_with_provider` is true and `external_customer_id` is given?
In this case, I think we should just use `external_customer_id` and create integration customer on Lago side.
That being said, probably we should have:
```
res = if external_customer_id.present?
    link_customer!
elsif sync_with_provider
    sync_customer!
end
```